AU2/database/AssassinsDatabase.py

```diff
@@ -44,12 +44,31 @@ def get(self, identifier: str) -> Assassin:
         """
         return self.assassins[identifier]
 
-    def get_identifiers(self) -> List[str]:
+    def get_filtered(self,
+                     filter: Callable[[Assassin], bool] = lambda x: True,
+                     include_hidden: Callable[[Assassin], bool] = lambda x: False) -> List[Assassin]:
+        """
+        Parameters:
+            filter: function that takes a non-hidden assassin and returns whether it should be included in the
+                returned list. Defaults to always return True.
+            include_hidden: function that takes a hidden Assassin object and returns whether it should be include in the
+                returned list. Defaults to always return False.
+
+        Returns:
+            list of Assassin objects satisfying the filter, excluding hidden assassins unless `include_hidden` is set to
+            True
+        """
+        return [a for a in self.assassins.values() if (a.hidden and include_hidden(a)) or (not a.hidden and filter(a))]
+
+    def get_identifiers(self, include_hidden: bool = False) -> List[str]:

```

Comment:
This function should be a wrapper around `get_filtered` - i.e., call it with a lambda to avoid code duplication.